### PR TITLE
add flow to check capacity configuration

### PR DESCRIPTION
**Pull request by AlanJager**
for https://github.com/zstackio/issues/issues/1945

### CODE REVIEW

File: plugin/kvm/src/main/java/org/zstack/kvm/KVMHost.java [hunk 1]
```diff
@@ -2374,6 +2375,21 @@ private void continueConnect(final boolean newAdded, final Completion completion
...
 
+        chain.then(new NoRollbackFlow() {
+            String __name__ = "check_cpu_and_memory_configuration";
+
+            @Override
+            public void run(FlowTrigger trigger, Map data) {
+                if (self.getCapacity().getTotalCpu() > SizeUtils.sizeStringToBytes(KVMGlobalConfig.RESERVED_MEMORY_CAPACITY.value())
+                        || self.getCapacity().getTotalMemory() > SizeUtils.sizeStringToBytes(KVMGlobalConfig.RESERVED_MEMORY_CAPACITY.value())) {
+                    trigger.fail(errf.stringToOperationError(String.format("The reserved capacity is not enough for adding a new host[uuid:%s]", self.getUuid())));
+                } else {
```
Comment (zxwing) [inline plugin/kvm/src/main/java/org/zstack/kvm/KVMHost.java]: “The reserved capacity is not enough for adding a new host[uuid:%s]”
改成：
"The host[uuid:%s] 's memory capacityp[%s] is lower than the minimal required capacity[%s]”

File: plugin/kvm/src/main/java/org/zstack/kvm/KVMHost.java [hunk 2]
```diff
@@ -2374,6 +2375,22 @@ private void continueConnect(final boolean newAdded, final Completion completion
             chain.then(extp.createKvmHostConnectingFlow(ctx));
         }
 
+        chain.then(new NoRollbackFlow() {
+            String __name__ = "check_cpu_and_memory_configuration";
+
+            @Override
+            public void run(FlowTrigger trigger, Map data) {
+                if (self.getCapacity().getTotalCpu() > SizeUtils.sizeStringToBytes(KVMGlobalConfig.RESERVED_MEMORY_CAPACITY.value())
```
Comment (zxwing) [inline plugin/kvm/src/main/java/org/zstack/kvm/KVMHost.java]: test

File: plugin/kvm/src/main/java/org/zstack/kvm/KVMHost.java [hunk 3]
```diff
@@ -2374,6 +2375,22 @@ private void continueConnect(final boolean newAdded, final Completion completion
             chain.then(extp.createKvmHostConnectingFlow(ctx));
         }
 
+        chain.then(new NoRollbackFlow() {
+            String __name__ = "check_cpu_and_memory_configuration";
+
+            @Override
+            public void run(FlowTrigger trigger, Map data) {
+                if (self.getCapacity().getTotalCpu() > SizeUtils.sizeStringToBytes(KVMGlobalConfig.RESERVED_CPU_CAPACITY.value())
```
Comment (zxwing) [inline plugin/kvm/src/main/java/org/zstack/kvm/KVMHost.java]: test

File: plugin/kvm/src/main/java/org/zstack/kvm/KVMHost.java [hunk 4]
```diff
@@ -2374,6 +2375,22 @@ private void continueConnect(final boolean newAdded, final Completion completion
...
+                        || self.getCapacity().getTotalMemory() > SizeUtils.sizeStringToBytes(KVMGlobalConfig.RESERVED_MEMORY_CAPACITY.value())) {
+                    trigger.fail(errf.stringToOperationError(String.format("The host[uuid:%s] 's memory capacity[%s] is lower than the minimal required capacity[%s]",
+                            self.getUuid(), self.getCapacity().getTotalMemory(), SizeUtils.sizeStringToBytes(KVMGlobalConfig.RESERVED_MEMORY_CAPACITY.value()))));
+                } else {
+                    new Log(self.getUuid()).log(HostLogLabel.CHECK_HOST_CPU_AND_MEMORY_CONFIGURATION);
+                    trigger.next();
+                }
+            }
+        });
+
```
Comment (Alvin-Lau) [inline plugin/kvm/src/main/java/org/zstack/kvm/KVMHost.java]: test

File: compute/src/main/java/org/zstack/compute/host/HostLogLabel.java [hunk 5]
```diff
@@ -19,6 +19,12 @@
     public static String ADD_HOST_CONNECT = "add.host.connect";
 
     @LogLabel(messages = {
+            "en_US = check cpu and memory configuration of the host",
+            "zh_CN = 检查物理机的cpu和内存配置"
+    })
+    public static String CHECK_HOST_CPU_AND_MEMORY_CONFIGURATION = "add.host.checkHostCpuAndMemoryConfiguration";
+
```
Comment (Alvin-Lau) [inline compute/src/main/java/org/zstack/compute/host/HostLogLabel.java]: test

File: plugin/kvm/src/main/java/org/zstack/kvm/KVMHost.java [hunk 6]
```diff
@@ -2374,6 +2375,22 @@ private void continueConnect(final boolean newAdded, final Completion completion
             chain.then(extp.createKvmHostConnectingFlow(ctx));
         }
 
+        chain.then(new NoRollbackFlow() {
+            String __name__ = "check_cpu_and_memory_configuration";
+
+            @Override
+            public void run(FlowTrigger trigger, Map data) {
+                if (self.getCapacity().getTotalCpu() > SizeUtils.sizeStringToBytes(KVMGlobalConfig.RESERVED_CPU_CAPACITY.value())
+                        || self.getCapacity().getTotalMemory() > SizeUtils.sizeStringToBytes(KVMGlobalConfig.RESERVED_MEMORY_CAPACITY.value())) {
```
Comment (zxwing) [inline plugin/kvm/src/main/java/org/zstack/kvm/KVMHost.java]: 写反了

**Pull request merged**